### ISSUE TITLE
Finesse Householder exit strategy

### ISSUE THREAD

**Issue by MikaelSlevinsky** [JuliaApproximation/ApproxFun.jl#481]
The last few Householder reflectors add spurious coefficients to the result that are usually below machine precision relative to the largest coefficients. There are some instances, for example when the result is numerically (anti-)symmetric, where the spurious coefficients aren't necessarily relatively small. Subsequent operations such as differentiation then incur larger errors by the act of raising the tail of coefficients. This issue is magnified by larger bandwidth problems.

**Comment by dlfivefifty** [comment]
I agree.  This is connected to a more general problem of plateaux detection.  A straightforward solution would be to copy the Chebfun plateaux detection algorithm, and see if it can be done adaptively.

Note that it should also be possible in O(n) to determine the "backward error" (right now it checks only the forward error) as is done in Olver's method.  I put backward error in quotes because it would only check the difference between the coefficients of  u_n and u_{n+b} I think for bandwidth b problems.

**Comment by MikaelSlevinsky** [comment]
I was thinking something like, "once we detect the number of degrees of freedom required for successful exit, backup and redo the number of Householder reflectors proportional to the lower bandwidth, but this time with truncation of the number of rows of the operator." This would more closely resemble exiting via Givens rotations, since the exit strategy was tested at every rotation.
